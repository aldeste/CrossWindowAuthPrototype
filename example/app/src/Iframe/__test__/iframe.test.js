import Iframe from "../Iframe";
import React from "react";
import renderer from "react-test-renderer";

jest.mock("../../Tags");

describe("Iframe component wrapper", () => {
  beforeAll(() => {
    global.window = jest.fn();
    global.window.addEventListener = jest.fn();
  });

  it("Exports", () => expect(renderer.create(<Iframe />)).toBeDefined());

  it("Renders as previously", () =>
    expect(renderer.create(<Iframe />).toJSON()).toMatchSnapshot());

  it("Updates height on message", () => {
    const tree = renderer.create(<Iframe />);

    expect(tree.toJSON().props.height).toBe("500px");
    tree.getInstance().heightManager({
      data: {
        type: "heightUpdate",
        data: {
          height: 1000
        }
      }
    });

    expect(tree.toJSON().props.height).toBe("1000px");
  });

  it("Ignores irelevant updates", () => {
    const tree = renderer.create(<Iframe />);

    expect(tree.toJSON().props.height).toBe("500px");
    tree.getInstance().heightManager({
      data: {
        type: "NOT_heightUpdate",
        data: {
          height: 1000
        }
      }
    });

    expect(tree.toJSON().props.height).toBe("500px");
  });
});
