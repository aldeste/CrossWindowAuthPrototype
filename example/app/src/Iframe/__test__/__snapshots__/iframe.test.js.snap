// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Iframe component wrapper Renders as previously 1`] = `
<iframe
  className="Iframe__Iframe-hXkCjZ HaLVb"
  height="500px"
/>
`;
